include/arch/arm/armv/armv8-a/64/armv/vcpu.h

```diff
@@ -26,7 +26,11 @@
 /* Allow native tasks to run at EL0, but restrict access */
 #define HCR_NATIVE ( HCR_COMMON | HCR_TGE | HCR_TVM | HCR_TTLB | HCR_DC \
                    | HCR_TAC | HCR_SWIO |  HCR_TSC | HCR_IMO | HCR_FMO | HCR_AMO)
+#ifdef CONFIG_PLAT_ZYNQMP
+#define HCR_VCPU   ( HCR_COMMON ) /* Allow zynqmp to use SMC calls */
```

Comment:
hi @chrisguikema I haven't put this commit as part of my updated 40-bit-pa support as I'm not sure it should go into master as is. Potentially we could hide it behind a config option? @kent-mcleod thoughts?

Reply:
Yeah, this should be hidden around a config option. Linux for the zynqmp uses a LOT of SMC calls to configure clocks and registers...obviously we don't want this behavior, but Linux can't run without it. I don't quite have the bandwidth for this, but I think the libsel4arm-vmm library will need to virtualize every SMC call. I think the config option is a good stop gap til that functionality exists.

Reply:
OK. Seems like this is a good solution for now.

As part of the work I'm doing at the moment I'll take a look at adding an `seL4_SMC_Control` cap to allow actual invocations.